grpc/recovery_test.go

```diff
@@ -1,28 +1,43 @@
-package server_test
+package grpc
 
 import (
+	"bytes"
 	"context"
+	"os"
+	"strings"
+	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 
-	. "github.com/contiamo/goserver/grpc"
 	"github.com/contiamo/goserver/grpc/test"
 )
 
 var _ = Describe("Recovery", func() {
 	It("should recover from panic when recovery option is set", func() {
+		buf := &bytes.Buffer{}
+		logrus.SetOutput(buf)
+		defer func() {
```

Comment:
I think it is probably safe to standardize this into a function that  always reset to info level and stdout, it will not really break anything, make the test environment more predictable and reduce the amount of boiler plate in the tests.

Reply:
I extracted the logging manipulation into `goserver/test/utils.go` also disabled logs in both `grpc` and `http` test suites except the buffering cases where we check the logs content in tests.

However, there are still two log entries on the console and they happen because some of the servers are still running after I run restore of the `DiscardLogging`. Probably the one in metrics `go goserver.ListenAndServeMetricsAndHealth(":8080", nil)` because it does not accept a context to cancel.

Reply:
Sounds like you found another trello ticket :)